File: 算法设计与分析作业/3.5 全排列问题.py
def Permutation(a,k,m):
    if k==m:
        for i in range(m+1):
            print(a[i],end=" ")
        print("")
    else:
        for i in range(k,m+1):
            a[i],a[k]=a[k],a[i]
            Permutation(a,k+1,m)
            a[i],a[k]=a[k],a[i]

a=[int(i) for i in input().split(" ")]
n=len(a)
a.sort()
Permutation(a,0,n-1)

File: 算法设计与分析作业/第2章 递归与分治/3.5 全排列问题.cpp
#include <iostream>
#include <cstdio>
#include <sstream>
#include <algorithm>
using namespace std;
int n,a[100005];
void swap(int &a,int &b)
{
	int t=a;
	a=b;
	b=t;
}
void Permutation(int a[],int k,int m)
{
	if(k==m)
	{
		for(int i=0;i<=m;i++)
			cout<<a[i]<<" ";
		cout<<endl;
	}
	else
	{
		for(int i=k;i<=m;i++)
		{
			swap(a[i],a[k]);
			Permutation(a,k+1,m);
			swap(a[i],a[k]);
		}
	}
}
int main()
{
	ios::sync_with_stdio(0),cin.tie(0),cout.tie(0);
	int t,pos=0;
	string s;
	stringstream ss;
	getline(cin,s);
	ss<<s;
	while(ss>>t)
		a[pos++]=t;
	n=pos;
	sort(a,a+n);
	Permutation(a,0,n-1);
	return 0;
}
